fineract-provider/src/main/java/org/apache/fineract/portfolio/search/service/SearchUtil.java

```diff
@@ -173,7 +232,9 @@ public static Object parseAndValidateColumnValue(final ResultsetColumnHeaderData
         if (StringUtils.isEmpty(columnValue)) {
             return columnValue;
         }
-        SQLInjectionValidator.validateDynamicQuery(columnValue);
+        if (strict) {
```

Comment:
strict: for stored procedures we do not want to validate the column values against injections to be inserted/updated/searched for.
sqlGenerator parameter instead of dialect, to fulfil @galovics request to avoid static methods on services.

Reply:
are we using/planning to use stored procedures?

Reply:
We are already using stored procedures on the newly implemented transaction and datatable get queries. Switch to stored procedure for datatable entry insert/update will come with the next PR.